### PR TITLE
feat: add grafana agent operator

### PR DESCRIPTION
**Pull request by gianklug**
# Description

Added the grafana agent operator to `misc-apps`. 

# Issues

# Checklist

* [x] This PR contains a description of the changes I'm making
* [x] I updated the version in Chart.yaml
* [x] I updated the changelog according to the `artifacthub.io/changes` annotation in the Chart.yaml. See following [example](https://github.com/adfinis/helm-charts/pull/858/files).
* [x] I updated applicable README.md files using  `pre-commit run`
* [x] I documented any high-level concepts I'm introducing in `docs/`
* [x] CI is currently green and this is ready for review
* [x] I am ready to test changes after they are applied and released

### CODE REVIEW

File: charts/misc-apps/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: misc-apps
 description: Argo CD app-of-apps config for miscellaneous small tools
 type: application
-version: 0.26.1
+version: 0.27.1
```
Comment (janaurka) [inline charts/misc-apps/Chart.yaml:5]: Why `.1`? Shouldn’t that just become `0.27.0`?

**Review by janaurka — CHANGES_REQUESTED** [review]
I’m okay with the thing except the version number which seems odd to me. Please elaborate (or change)

File: charts/misc-apps/Chart.yaml (hunk 1 above)
Comment (gianklug) [reply charts/misc-apps/Chart.yaml]: Sorry, my fault. Changed.

**Review by janaurka — APPROVED** [review]
LGTM!

**Pull request merged**